### PR TITLE
Restructure text and add alternative representation

### LINKED ISSUE #46: extend use beyond authorization servers
According to RFC7521 this draft aims to give specification about how to get authenticated with an AS, as we read in the introductory text

> [[RFC7521](https://vcstuff.github.io/draft-ietf-oauth-attestation-based-client-auth/draft-ietf-oauth-attestation-based-client-auth.html#RFC7521)] defines a way for a client to include an assertion in a token request to an authorization server for the purposes of client authentication. This specification uses this framework to define a new assertion type that provides a way for a client instance to authenticate itself with the authorization server through an assertion that is bound to a public key (for proof of possession).

If the usefulness and purpose of this draft is clear, at the same time we perceive a great potential that could also be useful in other contexts.

Lately there has been a lot of talk about the use of the nonce as a definitive prevention against replay attacks, and the importance of the nonce endpoint for issuing these is often discussed.

In the background we have the emerging wallet ecosystem which proposes reversed roles at different times, where the wallet is a client in the issuance phase towards a credential issuer which is an AS, while it is a sort of AS in the presentation phase towards the Relying Party.

This draft could enable the authentication of a client (wallet) to a third party (either this AS or RP) before obtaining the nonce.

the possible nonce endpoint could present itself as a protected endpoint, the nonce should only be issued to authenticated parties.

This generalization would require greater flexibility compared to the current purposes of the draft, which envisages its use only when the audience is an AS, while it could also be an RP, where the wallet instance attestation with PoP can be provided before any kind of transaction takes place (allowing the RP to discover the wallet capabilities before issuing the signed request object).

### LINKED ISSUE #76: Explicitly state the client attestation mechanism can be used at other endpoints such as PAR
Should we explcitly say that this may be used at the PAR endpoint?

_Originally posted by @paulbastian in https://github.com/vcstuff/draft-ietf-oauth-attestation-based-client-auth/pull/74#discussion_r1605361442_

### LINKED ISSUE #86: Restructure sections for presentation of attestation
My proposal would be:

- Section 5: HTTP header based syntax as currently defined but make clear that is not only for the token endpoint (merge 4.1 and 4.4 into this)
- Section 6: Alternative representation as SDJWT if header based syntax is not available (e.g., Wallet to RP case in openid4vp)

### PR DESCRIPTION
**Pull request by c2bo**
- fixes a small bug in the PoP examples (backslash)
- Restructures text into 2 major parts
  - header based syntax
  - concatenated representation (\<attestation jwt\> ~ \<attestation jwt pop\>)
- adds format & validation similar to an sd-jwt (but simpler)

Rendered version: https://drafts.oauth.net/draft-ietf-oauth-attestation-based-client-auth/c2bo/proposed-format/draft-ietf-oauth-attestation-based-client-auth.html

closes #86
closes #76
probably closes #46?

### CODE REVIEW

File: draft-ietf-oauth-attestation-based-client-auth.md [hunk 1]
```diff
@@ -259,6 +214,57 @@ The following example is the decoded header and payload of a JWT meeting the pro
...
+~~~
+OAuth-Client-Attestation       = token68
+OAuth-Client-Attestation-PoP   = token68
+token68                        = 1*( ALPHA / DIGIT / "-" / "." /
+                                     "_" / "~" / "+" / "/" ) *"="
+~~~
+
+It is RECOMMENDED that the authorization server validate the Client Attestation JWT prior to validating the Client Attestation PoP.
+
 ## Checking HTTP requests feature client attestations {#checking-http-requests-with-client-attestations}
```
Comment (paulbastian) [inline draft-ietf-oauth-attestation-based-client-auth.md:268]:
```suggestion
## Validating HTTP requests feature client attestations {#checking-http-requests-with-client-attestations}
```

File: draft-ietf-oauth-attestation-based-client-auth.md [hunk 2]
```diff
@@ -267,7 +273,7 @@ To validate an HTTP request which contains the client attestation headers, the r
 2. There is precisely one OAuth-Client-Attestation-PoP HTTP request header field, where its value is a single well-formed JWT conforming to the syntax outlined in []{client-attestation-pop-jwt}.
 3. The signature of the Client Attestation PoP JWT obtained from the OAuth-Client-Attestation-PoP HTTP header verifies with the Client Instance Key contained in the `cnf` claim of the Client Attestation JWT obtained from the OAuth-Client-Attestation HTTP header.
```
Comment (paulbastian) [inline draft-ietf-oauth-attestation-based-client-auth.md:274]:
```suggestion
2. There is precisely one OAuth-Client-Attestation-PoP HTTP request header field, where its value is a single well-formed JWT conforming to the syntax outlined in [](client-attestation-pop-jwt).
3. The signature of the Client Attestation PoP JWT obtained from the OAuth-Client-Attestation-PoP HTTP header verifies with the Client Instance Key contained in the `cnf` claim of the Client Attestation JWT obtained from the OAuth-Client-Attestation HTTP header.
```

File: draft-ietf-oauth-attestation-based-client-auth.md [hunk 3]
```diff
@@ -287,16 +293,90 @@ sIngiOiIxOHdITGVJZ1c5d1ZONlZEMVR4Z3BxeTJMc3pZa01mNko4bmpWQWlidmhNIiw\
...
+
+response_type=code&state=af0ifjsldkj&client_id=s6BhdRkqt3
+&redirect_uri=https%3A%2F%2Fclient.example.org%2Fcb
+&code_challenge=K2-ltc83acc4h0c9w6ESC_rEMTJ3bww-uCHaoeK1t8U
+&code_challenge_method=S256&scope=account-information
+~~~
+
+# Concatenated Serialization for Client Attestations {#alternative-representation}
+
+A Client Attestation according to this specification MAY be presented using an alternative representation for cases where the header-based mechanism (as introduced in introduced in []{#headers}) does not fit the underlying protocols, e.g., for direct calls to Browser APIs.
```
Comment (paulbastian) [inline draft-ietf-oauth-attestation-based-client-auth.md:340]:
```suggestion
A Client Attestation according to this specification MAY be presented using an alternative representation for cases where the header-based mechanism (as introduced in introduced in [](#headers) does not fit the underlying protocols, e.g., for direct calls to Browser APIs.
```

File: draft-ietf-oauth-attestation-based-client-auth.md [hunk 4]
```diff
@@ -287,16 +293,90 @@ sIngiOiIxOHdITGVJZ1c5d1ZONlZEMVR4Z3BxeTJMc3pZa01mNko4bmpWQWlidmhNIiw\
...
+iIDogIjVjMWE5ZTEwLTI5ZmYtNGMyYi1hZTczLTU3YzA5NTdjMDljNCIKfQ.coB_mtdX\
+wvi9RxSMzbIey8GVVQLv9qQrBUqmc1qj9Bs
+~~~
+
+## Validating the Concatenated Serialization {#validate-alternative}
+
+To validate a client attestation using the concatenated serialization form, the receiving server MUST ensure the following:
+
+1. Before the '~' character, there exists precisely a single well-formed JWT conforming to the syntax outlined in []{client-attestation-jwt}.
+2. After the '~' character, there exists precisely a single well-formed JWT conforming to the syntax outlined in []{client-attestation-pop-jwt}.
```
Comment (paulbastian) [inline draft-ietf-oauth-attestation-based-client-auth.md:377]:
```suggestion
1. Before the '~' character, there exists precisely a single well-formed JWT conforming to the syntax outlined in [](client-attestation-jwt).
2. After the '~' character, there exists precisely a single well-formed JWT conforming to the syntax outlined in [](client-attestation-pop-jwt).
```

**Review by paulbastian — APPROVED** [review]
lgtm, should we include that the header based syntax is a MUST implementation support?

**Comment by c2bo** [comment]
> lgtm, should we include that the header based syntax is a MUST implementation support?

I thought about language like that as well, but I don't think it really helps here. It should be in something like HAIP that profiles an ecosystem - a client will/must know what it needs to support and an AS etc. can choose depending on the ecosystem.

File: draft-ietf-oauth-attestation-based-client-auth.md [hunk 5]
```diff
@@ -259,15 +214,66 @@ The following example is the decoded header and payload of a JWT meeting the pro
 }
 ~~~
 
-## Checking HTTP requests feature client attestations {#checking-http-requests-with-client-attestations}
+# Client Attestation using a Header based syntax
+
+The default way to present a Client Attestation is by using the header-based syntax that can be used at different endpoints.
```
Comment (tplooker) [inline draft-ietf-oauth-attestation-based-client-auth.md:219]:
```suggestion
The following section defines how a Client Attestation can be provided in an HTTP request using HTTP headers.
```

Comment (tplooker) [reply draft-ietf-oauth-attestation-based-client-auth.md:219]: No real notion of default here so thought it was better to rephrase.

**Review by tplooker — APPROVED** [review]

**Pull request merged**